### DISCUSSION TITLE
How to run both regular JUnit Unit tests and QuarkusTest?

### DISCUSSION THREAD

**Discussion by tmulle** [quarkusio/quarkus#39622] [Q&A]
Hi, sorry if this has been answered but I'm trying to run both standard JUnit Unit tests as well QuarkusTests and have a jacoco report generated for both.

Right now, When I mark my plain Unit test (no test containers, db, etc. needed) with `@QuarkusTest` along with the other test that DOES need the DB test container and is marked `@QuarkusTest` and run `mvn clean verify`, both test classes are run and the jacoco report is generated.

BUT each test class spins up and tears down the DB after each run. I don't need the DB to start when I'm just trying to test utility classes, etc.

If I remove the `@QuarkusTest` from my standard Junit class and run `mvn test` then the dev services DO NOT start and the test runs, but then the jacoco reports don't get generated.

I've looked at https://quarkus.io/guides/tests-with-coverage#prerequisites and have the settings in my pom.xml they mentioned.

Hope this makes sense, I can send more information as needed.

**Comment by ennishol** [comment]
The easiest approach is to name your plain tests *Test.java and the tests with DB *IT.java. By using surefire and failsafe plugins they will be separated. You can decide if you want to have code coverage to be combined from both plugins or not.
Also, have a look at [@QuarkusComponentTest]( https://quarkus.io/blog/quarkus-component-test/).
Hope this helps

**Comment by jdboer-dhl** [comment]
I think this is now covered by the documentation for [Coverage for tests not using @QuarkusTest](https://quarkus.io/guides/tests-with-coverage#coverage-for-tests-not-using-quarkustest)
